e2e/nomostest/iam/account.go

```diff
@@ -39,3 +44,32 @@ func DescribeServiceAccount(nt *nomostest.NT, gsaEmail, projectID string) ([]byt
 	}
 	return out, nil
 }
+
+func getWIUser(fleetProject string, ksaRef types.NamespacedName) string {
+	return fmt.Sprintf("%s.svc.id.goog[%s/%s]", fleetProject, ksaRef.Namespace, ksaRef.Name)
+}
+
+// ImpersonateGSA impersonates a service account by binding the KSA to GSA.
+func ImpersonateGSA(nt *nomostest.NT, fleetProject, gsaEmail string, ksaRef types.NamespacedName) error {
+	_, err := iamPolicyBinding(nt, "add-iam-policy-binding", gsaEmail, WIUserRole, getWIUser(fleetProject, ksaRef))
+	if err != nil {
+		return fmt.Errorf("adding IAM policy binding to an IAM service account: %w", err)
+	}
+	return nil
+}
+
+// UnimpersonateGSA removes the IAM policy binding from the IAM service account
+func UnimpersonateGSA(nt *nomostest.NT, fleetProject, gsaEmail string, ksaRef types.NamespacedName) error {
+	out, err := iamPolicyBinding(nt, "remove-iam-policy-binding", gsaEmail, WIUserRole, getWIUser(fleetProject, ksaRef))
+	if err != nil && !strings.Contains(string(out), "Policy binding with the specified principal, role, and condition not found") {
```

Comment:
Output is subject to change and could break our tests. Does it provide a flag for ignoring not found errors?

Reply:
Unfortunately, it doesn't provide such a flag. Another option is to only check for the `not found` string, but it seems not very specific.

Reply:
That's fine. We'll need to update test code if the gcloud output changes and breaks us.